src/network/mod.rs

```diff
@@ -30,3 +34,44 @@ pub mod message;
 pub mod message_blockdata;
 pub mod message_network;
 
+/// Network error
+#[derive(Debug)]
+pub enum Error {
+    /// And I/O error
+    Io(io::Error),
+    /// Socket mutex was poisoned
+    SocketMutexPoisoned,
+    /// Not connected to peer
+    SocketNotConnectedToPeer,
+    /// Error propagated from subsystem
+    Detail(String, Box<Error>),
```

Comment:
Variant can be dropped (see comments on network/listener.rs which is the only place this was used)